poglink/bot.py

```diff
@@ -32,16 +33,23 @@ def __init__(
         self.rates_urls = rates_urls
         # self.bans_url = bans_url
         self.data_dir = data_dir
+        self.publish_on_startup = publish_on_startup
+
+        # handle special cases for boolean values
+        for param in BOOLEAN_VALUES:
+            if not isinstance(getattr(self, param), bool):
+                setattr(self, param, parse_bool(getattr(self, param)))
 
         # handle special cases for list parsing
-        for val in LIST_VALUES:
-            if isinstance(getattr(self, val), str):
-                try:
-                    setattr(self, val, parse_list(getattr(self, val)))
-                except TypeError as e:
-                    logger.warning(
-                        f"Incorrect variable format for {val}; should be comma separated list. Actual value: {config[val]}; {e}"
-                    )
+        for param in LIST_VALUES:
+            if isinstance(getattr(self, param), str):
+                setattr(self, param, parse_list(getattr(self, param)))
+            elif isinstance(getattr(self, param), list):
+                pass
+            else:
+                logger.warning(
+                    f"Incorrect variable format for {param}; should be comma separated list. Actual value: {getattr(self, param)}"
+                )
```

Comment:
The exception previously being caught here is impossible to ever have raised. It only tries to `parse_list` if the input type is `str`. But `parse_list` cannot fail on a `str`. So instead I'm just raising the same warning when the input is not a list or string.